src/app/containers/LoginPage/index.js

```diff
@@ -4,123 +4,124 @@ import React from 'react';
 import { apiLogin } from '../../../api/authentication';
 
 // HOC
-import { withRouter } from 'react-router-dom';
+import { withRouter, Redirect } from 'react-router-dom';
 import withUserNotLogin from '../../../shared/hoc/withUserNotLogin';
+import { UserInfoContext } from '../../../shared/context/UserInfo';
 
-// Component
-import PopupFailed from './PopupFailed';
-import InputText from '../../components/InputText';
+// UI
+import * as MainPanel from '../../common-ui/MainPanel';
+import * as Form from '../../common-ui/Form';
+import LabeledInput from '../../common-ui/LabeledInput';
+import * as Button from '../../common-ui/Button';
+import Popup from '../../common-ui/Popup';
 
 // Constants
 import { API_PROGRESS } from '../../../shared/constants';
 
-import styles from './login.scss';
+function LoginPage() {
+  const [apiProgress, setApiProgress] = React.useState(API_PROGRESS.INIT);
+  const { userInfo } = React.useContext(UserInfoContext);
+  const [data, setData] = React.useState({
+    // null: pristine (user has not changed the value)
+    // empty string: non-pristine (user has changed the value)
+    username: '',
+    password: '',
+  });
 
-class LoginPage extends React.Component {
-  constructor() {
-    super();
-    this.state = {
-      username: '',
-      password: '',
-      showFalsePopup: false,
-      apiProgress: API_PROGRESS.INIT,
-    };
-    this.handleSubmit = this.handleSubmit.bind(this);
-    this.handleChange = this.handleChange.bind(this);
-    this.closePopup = this.closePopup.bind(this);
-  }
+  const [popupState, setPopupState] = React.useState(false);
+  const [redirectState, setRedirectState] = React.useState(false);
 
-  closePopup() {
-    this.setState({ showFalsePopup: false });
-  }
+  const handleChange = React.useCallback((name, value) => {
+    setData({ ...data, [name]: value });
+  });
```

Comment:
Missing the second param for `useCallBack`, either add it an empty array or just remove `useCallback` no need useCallback here for just a simple onChange function